erpnext/accounts/report/general_ledger/general_ledger.js

```diff
@@ -53,7 +53,9 @@ frappe.query_reports["General Ledger"] = {
 			"label": __("Voucher No"),
 			"fieldtype": "Data",
 			on_change: function() {
-				frappe.query_report.set_filter_value('group_by', "");
+				if (!this.value) {
+					frappe.query_report.set_filter_value('group_by', "");
+				}
```

Comment:
This does not make any sense, instead on change of voucher_no set group_by as "Group by Voucher (Consolidated)"

```suggestion
				frappe.query_report.set_filter_value('group_by', "Group by Voucher (Consolidated)");
```